.github/workflows/ci.yml

```diff
@@ -165,6 +173,30 @@ jobs:
               ../build/examples/trixi_controller_data_f . libelixir_p4est2d_dgsem_euler_sedov.jl
         env:
           LIBTRIXI_DEBUG: all
+      - name: Check error handling of examples
+        if: ${{ matrix.test_type == 'coverage' }}
+        continue-on-error: true
+        run: |
+          cd libtrixi-julia
+          ../build/examples/simple_trixi_controller_c
```

Comment:
Concerning the examples:

We can do something like this to increase coverage of the examples.
But:
1. I could not quickly figure out how to continue execution after the first command has failed
2. It would be nice to check of the return codes are indeed 2 as expected
3. Isn't this overkill?

Reply:
> 1. I could not quickly figure out how to continue execution after the first command has failed
> 2. It would be nice to check of the return codes are indeed 2 as expected

The bash shell in GitHub Actions [by default](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) runs with `set -e`, i.e., the first non-zero return value of any command causes to exit prematurely and mark the step as a failure.

What you could try is to manually catch and compare the exit code, e.g., something like this
```suggestion
      - name: Check error handling of examples
        if: ${{ matrix.test_type == 'coverage' }}
        run: |
          cd libtrixi-julia
          set +e # disable early exit on non-zero exit code
          ../build/examples/simple_trixi_controller_c
          if [ $? -eq 2 ]; then
            echo "Correct exit code 👍"
            exit 0
          else
            echo "ERROR: expected exit code 2" >&2
            exit 1
          fi
```
Note that I have not yet tested the code, so please verify it works.

> 3. Isn't this overkill?

I don't know. Maybe it is, but OTOH, we want to keep coverage up up up 😬

Reply:
Great, thanks for the hints! So it is save to assume bash and we do not need to be cross platform compatible?

Reply:
In libtrixi, we explicitly only support Linux at the moment, so I think it's ok. `bash` is also the default for GHA workflows, if you look carefully in the link to the GH docs above. Otherwise, you can also explicitly specify `bash` as the shell, which works cross platform (though I doubt we will ever support Windows in the foreseeable future).

Reply:
It worked!